hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/ServerRemoteProcedure.java

```diff
@@ -80,6 +81,8 @@ public abstract class ServerRemoteProcedure extends Procedure<MasterProcedureEnv
   protected ServerName targetServer;
   protected boolean dispatched;
   protected boolean succ;
+  protected MasterProcedureProtos.ServerRemoteProcedureState state =
```

Comment:
If I am using state for making decisions then I think we need to handle HMaster restart scenarios while HBase upgrade.